react/features/chat/components/web/ChatMessage.tsx

```diff
@@ -167,42 +255,151 @@ const ChatMessage = ({
         );
     }
 
+    /**
+     * Renders the reactions for the message.
+     *
+     * @returns {React$Element<*>}
+     */
+    function renderReactions() {
+        if (!message.reactions || message.reactions.size === 0) {
+            return null;
+        }
+
+        const reactionsArray = Array.from(message.reactions.entries())
+            .map(([ reaction, participants ]) => {
+                return { reaction,
+                    participants };
+            })
+            .sort((a, b) => b.participants.size - a.participants.size);
+
+        const openReactions = Boolean(reactionsAnchorEl);
+        const reactionsId = openReactions ? 'reactions-popover' : undefined;
+
+        return (
+            <>
+                <div
+                    className = { classes.reactionBox }
+                    onClick = { handleReactionsClick }>
+                    {reactionsArray.slice(0, 3).map(({ reaction }, index) => (
+                        <span key = { index }>
+                            {reaction}
+                        </span>
+                    ))}
+                    {reactionsArray.length > 3 && (
+                        <span className = { classes.reactionCount }>
+                            +{reactionsArray.length - 3}
```

Comment:
I think we should count the number of reactions, not the number of participants that reacted with various reactions, wdys?

Reply:
WhatsApp counts the total reactions, FWIW.